routing/endpointregistry.go

```diff
@@ -115,16 +150,69 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 	return routes
 }
 
+func (r *EndpointRegistry) updateStats() {
+	ticker := time.NewTicker(r.statsResetPeriod)
+
+	for {
+		r.data.Range(func(key, value any) bool {
+			e := value.(*entry)
+
+			curSlot := e.curSlot.Load()
+			nextSlot := (curSlot + 1) % 2
+			e.totalFailedRoundTrips[nextSlot].Store(0)
+			e.totalRequests[nextSlot].Store(0)
+
+			failed := e.totalFailedRoundTrips[curSlot].Load()
+			requests := e.totalRequests[curSlot].Load()
+			failedRoundTripsRatio := float64(failed) / float64(requests)
```

Comment:
Is division by zero possible here?

Reply:
Yes, I must move this code under if block below.

Reply:
The division by zero is possible, it will make failedRoundTripsRatio equal to 'NaN', and this will make drop probability equal to None. That will cause no problems by itself, but it is very implicit, of course I need:
+ move this code under if block to avoid NaNs
+ write a test that will check PHC won't crash if there're no requests during period (to be on the safe side).